### DISCUSSION TITLE
close_range syscall in container returns operatio not permitted

### DISCUSSION THREAD

**Discussion by carrotIndustries** [community/community#12795] [Actions]
Hello there,

since [recently](https://gitlab.gnome.org/GNOME/glib/-/commit/ce04a124040be091407e070280d86ca810bacb8c), Glib actually checks the result of the `close_range` syscall and errors out if it failed. 

Since [other people](https://gitlab.gnome.org/Infrastructure/GitLab/-/issues/545) were also having issues in their CI setups, I assume that there's also something wrong with GitHub Actions that causes the `close_range` syscall to return permission denied: https://github.com/horizon-eda/horizon/runs/5529896277?check_suite_focus=true

I was able to work around this with and LD_PRELOAD kludge that makes `close_range` return `ENOSYS`, but it'd be nice if the syscall  either returned ENOSYS or did the right thing.

If someone can tell me which part of Actions is at fault here, I can file an issue there.

**Comment by ethomson** [comment]
There's nothing clever here that GitHub Actions is doing. Certainly there could be a problem with the kernel, but my first question would be whether you see these errors on a run in GitHub Actions directly on our virtual machines? Or only when running in Docker?

**Reply by carrotIndustries** [reply]
> directly on our virtual machines? Or only when running in Docker?

I ran some tests:

 - Running on without docker seems to work: https://github.com/carrotIndustries/close-range-test/runs/5779788820?check_suite_focus=true#step:4:21 the particular version of strace that ships with ubuntu 20.04 doesn't know about the syscall, but it seems to succeed 
 - Running in docker, the syscall returns "operation not permitted": https://github.com/carrotIndustries/close-range-test/runs/5779788751?check_suite_focus=true#step:5:20 and https://github.com/carrotIndustries/close-range-test/runs/5779788862?check_suite_focus=true#step:5:20
